File: week5/py/question_paper.py
# ///https://codeforces.com/edu/course/2/lesson/6 for explanations of bs problems
t=int(input())

for _ in range(t):
    n,m=list(map(int,input().split()))
    floors=list(map(int,input().split()))
    bundle=list(map(int,input().split()))
    for i in range(1,n):
        floors[i]+=floors[i-1]
    for k in bundle:
        l=-1
        r=n
        #equivalent to c++ lower_bound()
        while(l+1<r):
            mid = (l+r)//2
            if (floors[mid]>=k):
                r=mid
            else:
                l=mid
        print(r+1,end=' ')
    print()

File: week5/cpp/questionpaper.cpp
///https://codeforces.com/edu/course/2/lesson/6 for explanations of bs problems
#include <bits/stdc++.h> // Include all standard C++ libraries
using namespace std;     // Use the standard namespace

void solve() {

  long long n, m;
  cin >> n >> m;
  long long a[n];
  for (int i = 0; i < n; i++)
    cin >> a[i];
  for (int i = 1; i < n; i++)
    a[i] += a[i - 1];

  while (m--) {

    long long x;
    cin >> x;

    cout << ((lower_bound(a, a + n, x) - a) + 1) << " ";
  }

  cout << "\n";
}

int main() {

  int tstCnt; // Declare an integer variable for the number of test cases

  cin >> tstCnt; // Read the number of test cases from the input

  while (tstCnt--) { // Loop for each test case

    solve(); // Call the solve function
  }

  return 0;
}
